File: src/main/java/com/Wurger/model/UnidadMedida.java
package com.Wurger.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "unidad_medida")
@Getter
@Setter
public class UnidadMedida {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_unidad")
    private Integer id;

    @Column(nullable = false, length = 50)
    private String nombre;

    private Integer cantidad;

    // Relación Muchos a Uno con Producto
    @ManyToOne
    @JoinColumn(name = "id_producto")
    private Producto producto;
}

File: app/Models/UnidadMedida.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UnidadMedida extends Model
{
    use HasFactory;

    protected $table = 'unidad_medida';
    protected $primaryKey = 'id_unidad';

    protected $fillable = [
        'nombre',
        'cantidad',
        'id_producto'
    ];

    public function producto()
    {
        return $this->belongsTo(Producto::class, 'id_producto', 'id_producto');
    }
}
